CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+# Contributing to Charon
+
+First off, thanks for taking the time to contribute!
+
+All types of contributions are encouraged. See the [Table of Contents](#table-of-contents) for different ways to help. Please make sure to read the relevant section before making your contribution. We look forward to your contributions!
+
+## Table of Contents
+
+- [I Have a Question](#i-have-a-question)
+- [How To Contribute](#how-to-contribute)
+- [Reporting Bugs](#reporting-bugs)
+- [Suggesting Enhancements](#suggesting-enhancements)
+- [Contributing Code](#contributing-code)
+
+
+## I Have a Question
+
+<!--
+> If you want to ask a question, we assume that you have read the available [Documentation]().
+-->
+
+Before you ask a question, it is best to search for existing [Issues](https://github.com/AeneasVerif/charon/issues) that might help you. In case you have found a suitable issue and still need clarification, you can write your question in this issue.
+
+If you then still feel the need to ask a question and need clarification, you can either:
+- ask your question on our [Zulip](https://aeneas-verif.zulipchat.com/);
+- or open an [Issue](https://github.com/AeneasVerif/charon/issues/new) (see [below](#reporting-bugs) for how to report a bug).
+
+## How To Contribute
+
+### Reporting Bugs
+
+#### Before Submitting a Bug Report
+
+To make our lives as maintainers as easy as possible, we ask you to investigate carefully to isolate the cause of your issue. You may want to check:
+
+- Make sure that you are using the latest version of Charon;
+- Ensure you're using the same version of Charon to produce and consume llbc files;
+- Clean up cached files and recompile (inside the project, this can be done with `make clean; make`);
+- If relevant, recompile the project you use to consume llbc files (e.g. Aeneas of Eurydice);
+- Clean up leftover llbc files and regenerate them (inside the project, this can be done with `make clean-generated`);
+- Check if a similar issue has already been reported in the [issue tracker][].
+
+#### Reporting a Bug
+
+To be able to help you, we need the following information:
```

Comment:
We should put this in an issue template

Reply:
Tracker in https://github.com/AeneasVerif/charon/issues/198